### PR TITLE
[WNMGDS-691] Add guidance for help drawer

### PR DESCRIPTION
**Pull request by nichia**
### Added
- Add guidance for help drawer usage
- Refer to [Help drawer link enhancements efforts](https://miro.com/app/board/o9J_kklLajE=/) for further iteration of the guidance content and [help drawer links research](https://miro.com/app/board/o9J_kqbwNQg=/)

### How to test
- Run the doc site locally
- Visit the help drawer page to check the updated guidance

### CODE REVIEW

File: packages/design-system-docs/src/pages/components/HelpDrawer/_HelpDrawer.docs.scss [hunk 1]
```diff
@@ -33,3 +33,43 @@ Style guide: components.help-drawer.react-help-drawer
...
+/*
+---
+
+### When to use
+
+- When users need help content to remain viewable while also allowing the user to views/interacts with the main page.
+- When the help content is longer than 2 sentences.
+- When the content needs to be available at a particular point in the process, but doesn't need to be included on the main page.
+- When the content just needs to be available at the user's discretion, to provide additional details or context.
+- When users may need additional information to complete the task than can be supplied on the main page without overwelming the primary content.
```
Comment (bernardwang) [inline packages/design-system-docs/src/pages/components/HelpDrawer/_HelpDrawer.docs.scss:46]: overwhelming

File: packages/design-system-docs/src/pages/components/HelpDrawer/_HelpDrawer.docs.scss [hunk 2]
```diff
@@ -33,3 +33,43 @@ Style guide: components.help-drawer.react-help-drawer
...
+
+- When the content is timely/urgent, or critical to the action being taken.
+- When the content is intended to clarify or call attention to other content on the page.
+- When the content must be displayed without being initiated by a user action.
+- When additional information is necessary to the primary information the user is interacting with.
+
+### Usage
+
+- Link text used to trigger the help drawer should be specific and descriptive.
+  - Avoid using "Learn more"
```
Comment (bernardwang) [inline packages/design-system-docs/src/pages/components/HelpDrawer/_HelpDrawer.docs.scss:58]: Is there an extra space here?

**Review by bernardwang — APPROVED** [review]

File: packages/design-system-docs/src/pages/components/HelpDrawer/_HelpDrawer.docs.scss [hunk 3]
```diff
@@ -33,3 +33,43 @@ Style guide: components.help-drawer.react-help-drawer
...
+  - Limit the amount of link text
+- Help drawer content area should include a close button clearly at the top.
+
+### Accessibility
+
+- Focus should move to the content in the help drawer after user has opened it.
```
Comment (line47) [inline packages/design-system-docs/src/pages/components/HelpDrawer/_HelpDrawer.docs.scss:64]:
```suggestion
- Focus should move to the content in the help drawer after the user has opened it.
```

File: packages/design-system-docs/src/pages/components/HelpDrawer/_HelpDrawer.docs.scss (hunk 2 above)
Comment (line47) [inline packages/design-system-docs/src/pages/components/HelpDrawer/_HelpDrawer.docs.scss:58]:
```suggestion
  - Avoid using only "Learn more" to link to help drawer content
```

**Review by line47 — DISMISSED** [review]
A few small word changes but otherwise looks good, thanks for adding!

File: packages/design-system-docs/src/pages/components/HelpDrawer/_HelpDrawer.docs.scss (hunk 1 above)
Comment (nichia) [inline packages/design-system-docs/src/pages/components/HelpDrawer/_HelpDrawer.docs.scss:46]:
```suggestion
- When users may need additional information to complete the task than can be supplied on the main page without overwhelming the primary content.
```

**Pull request merged**